### PR TITLE
feat: jsonL support

### PR DESCRIPTION
**Pull request by Jorjeous**
### PR fulfills these requirements
- [+] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [-] Tests for the changes have been added/updated (for bug fixes/features)
- [-] Docs have been added/updated (for bug fixes/features)
- [-] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [+] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [ ] Frontend
- [+] OtherBackend

### Describe the reason for change
_(link to issue, supportive screenshots etc.)_
People often confuse json and jsonL
The nemo framework deliberately uses the name json, although in fact it is jsonL

#### What does this fix?
_(if this is a bug fix)_

#### What is the new behavior?
_(if this is a breaking or feature change)_
Ability to import jsonL

#### What is the current behavior?
_(if this is a breaking or feature change)_

#### What libraries were added/updated?
_(list all with version changes)_
No 

#### Does this change affect performance?
_(if so describe the impacts positive or negative)_
No

#### Does this change affect security?
_(if so describe the impacts positive or negative)_
No

#### What alternative approaches were there?
_(briefly list any if applicable)_

#### What feature flags were used to cover this change?
_(briefly list any if applicable)_

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [+] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit
- [+] Manual

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_
Data import

### CODE REVIEW

**Comment by sajarin** [comment]
/jira create
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9007382438)
> Jira issue [TRIAG-550](https://humansignal.atlassian.net/browse/TRIAG-550) is created

[TRIAG-550]: https://humansignal.atlassian.net/browse/TRIAG-550?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by Jorjeous** [comment]
After that pr I'm going to add 
1) some new default templates (Automatic Speech Recognition (ASR) with ground truth (GT),  ASR + GT + model prediction )
2) scripts for cutting long audios to small jobs using NFA (Nemo Forced Aligner) 
3) template for long audio, that utilizes ML backend, and java script, that follows text (according to the audio)

**Comment by sajarin** [comment]
thanks for the PR @Jorjeous, our team is working through some PRs and we'll have a review for this one soon, thanks for being patient! 

related reports for jsonL support: 
https://github.com/HumanSignal/label-studio-converter/issues/58
https://github.com/HumanSignal/label-studio/issues/1960
https://github.com/HumanSignal/label-studio/issues/4607

File: label_studio/data_import/models.py [hunk 1]
```diff
@@ -109,6 +123,35 @@ def read_tasks_list_from_json(self):
...
+        from .uploader import create_file_upload
+        
+        logger.debug('Read tasks list from JSONL file {}'.format(self.file.name))
+        content = self.content.splitlines()
+        tasks = []
+
+        for line in content:
+            task_data = json.loads(line)
+    
+            if 'audio_filepath' in task_data:
```
Comment (makseq) [inline label_studio/data_import/models.py:137]: what is it? It shouldn't be depended on any specific keys in task.data

File: label_studio/data_import/models.py [hunk 2]
```diff
@@ -109,6 +123,35 @@ def read_tasks_list_from_json(self):
...
+                    # Get the path of the audio file
+                    audio_path = task_data.pop('audio_filepath')  # pop - to remove the original key
+                    
+                    if os.path.exists(audio_path):
+                        # Upload audio to LS file storage
+                        with open(audio_path, 'rb') as audio_file:
+                            # Create a FileUpload for the audio
+                            audio_upload = create_file_upload(self.user, self.project, SimpleUploadedFile(os.path.basename(audio_path), audio_file.read()))
```
Comment (makseq) [inline label_studio/data_import/models.py:145]: It most cases LS doesn't have an access to audio_path, because it's deployed on a separated machine/docker

**Review by makseq — CHANGES_REQUESTED** [review]

**Comment by robot-ci-heartex** [comment]
This PR is stale because it has been open 45 days with no activity. Remove `stale` label or comment or this will be closed in 10 days.

**Comment by robot-ci-heartex** [comment]
This PR was closed because it has been stalled for 10 days with no activity.

**Pull request closed without merging**